0.12.2 does not work with repos with a lot of pull requests

First I want to thank you for creating this! I have been using this daily for my work since version 0.9 and it has been great!

Recently I upgraded to 0.12.2 and it seems the new way of pulling PRs does not work very well for repos with a lot of PRs open. I kept getting errors like the following

```
 Date: Fri May 12 14:26:27 2023 -0700
 6 files changed, 70 insertions(+), 23 deletions(-)
> git rev-parse --show-toplevel
> git fetch
> git branch --no-color
> git rebase origin/master --autostash
> github fetch pull requests
> git branch --no-color
> git branch --no-color
> git log --format=medium --no-color origin/master..HEAD
> git rebase origin/master -i --autosquash --autostash
> git log --format=medium --no-color origin/master..HEAD
> git branch --no-color
> git branch --no-color
> git log --format=medium --no-color origin/master..HEAD
> git status --porcelain --untracked-files=no
> git push --force --atomic origin 48cf23e2eda59f3348198ae2f16942eade3e69ae:refs/heads/spr/ed125cd8
> git branch --no-color
> github create 9387 : Make import under try optional
> github update 9387 : Make import under try optional
> git branch --no-color
> github fetch pull requests
> git branch --no-color
> git branch --no-color
> git log --format=medium --no-color origin/master..HEAD
> git branch --no-color
pull request stack is empty
```

Is it possible to bring back the old way of retrieving PRs? Or maybe control it with some flag?

BTW it would be great if spr can ignore unrecognized flags in `.spr.yml` under a repo. Otherwise, when new flags are added, people using old versions of spr are either forced to upgrade or force git to ignore changes to `.spr.yml`. Neither is great.

Also, sometimes `git spr update` fails with
```
> git rev-parse --show-toplevel
> git fetch
git error: error: cannot lock ref 'refs/remotes/origin/master': is at c2b5bd07d8dba653b26a6858a2fb40e9815be222 but expected 3d26e34b51496570a2b633b6f66c40bad635160f
From github.com:pinternal/pinboard
 ! 3d26e34b5149..c2b5bd07d8db  master     -> origin/master  (unable to update local ref)
panic: exit status 1

goroutine 1 [running]:
github.com/ejoffe/spr/git/realgit.(*gitcmd).MustGit(0x101000000000002, {0x88ee68, 0x0}, 0x8)
        /Users/runner/work/spr/spr/git/realgit/realcmd.go:44 +0x45
github.com/ejoffe/spr/spr.(*stackediff).fetchAndGetGitHubInfo(0xc0000b9020, {0x92c590, 0xc0000a8000})
        /Users/runner/work/spr/spr/spr/spr.go:468 +0x8d
github.com/ejoffe/spr/spr.(*stackediff).UpdatePullRequests(0xc0000b9020, {0x92c590, 0xc0000a8000}, {0x0, 0x4, 0x0}, 0x0)
        /Users/runner/work/spr/spr/spr/spr.go:119 +0xa5
main.main.func4(0xc0000d8b00)
        /Users/runner/work/spr/spr/cmd/spr/main.go:155 +0x114
github.com/urfave/cli/v2.(*Command).Run(0xc0001ac900, 0xc0000bb3c0)
        /Users/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/command.go:163 +0x64a
github.com/urfave/cli/v2.(*App).RunContext(0xc0001b6340, {0x92c590, 0xc0000a8000}, {0xc0000bc000, 0x2, 0x2})
        /Users/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:313 +0x81e
github.com/urfave/cli/v2.(*App).Run(...)
        /Users/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:224
main.main()
        /Users/runner/work/spr/spr/cmd/spr/main.go:224 +0x1325
```

--- a/github/githubclient/client.go
+++ b/github/githubclient/client.go
@@ -185,7 +185,7 @@ func (c *client) GetInfo(ctx context.Context, gitcmd git.GitInterface) *github.G
 	targetBranch := c.config.Internal.GitHubBranch
 	localCommitStack := git.GetLocalCommitStack(c.config, gitcmd)
 
-	pullRequests := matchPullRequestStack(c.config.Repo, targetBranch, localCommitStack, resp.Repository.PullRequests)
+	pullRequests := matchPullRequestStack(c.config.Repo, targetBranch, localCommitStack, resp.Viewer.PullRequests)
 	for _, pr := range pullRequests {
 		if pr.Ready(c.config) {
 			pr.MergeStatus.Stacked = true
@@ -209,7 +209,7 @@ func matchPullRequestStack(
 	repoConfig *config.RepoConfig,
 	targetBranch string,
 	localCommitStack []git.Commit,
-	allPullRequests genclient.PullRequestsRepositoryPullRequests) []*github.PullRequest {
+	allPullRequests genclient.PullRequestsViewerPullRequests) []*github.PullRequest {
 
 	if len(localCommitStack) == 0 || allPullRequests.Nodes == nil {
 		return []*github.PullRequest{}

--- a/github/githubclient/client_test.go
+++ b/github/githubclient/client_test.go
@@ -15,19 +15,19 @@ func TestMatchPullRequestStack(t *testing.T) {
 	tests := []struct {
 		name    string
 		commits []git.Commit
-		prs     genclient.PullRequestsRepositoryPullRequests
+		prs     genclient.PullRequestsViewerPullRequests
 		expect  []*github.PullRequest
 	}{
 		{
 			name:    "Empty",
 			commits: []git.Commit{},
-			prs:     genclient.PullRequestsRepositoryPullRequests{},
+			prs:     genclient.PullRequestsViewerPullRequests{},
 			expect:  []*github.PullRequest{},
 		},
 		{
 			name:    "FirstCommit",
 			commits: []git.Commit{{CommitID: "00000001"}},
-			prs:     genclient.PullRequestsRepositoryPullRequests{},
+			prs:     genclient.PullRequestsViewerPullRequests{},
 			expect:  []*github.PullRequest{},
 		},
 		{
@@ -36,16 +36,16 @@ func TestMatchPullRequestStack(t *testing.T) {
 				{CommitID: "00000001"},
 				{CommitID: "00000002"},
 			},
-			prs: genclient.PullRequestsRepositoryPullRequests{
-				Nodes: &genclient.PullRequestsRepositoryPullRequestsNodes{
+			prs: genclient.PullRequestsViewerPullRequests{
+				Nodes: &genclient.PullRequestsViewerPullRequestsNodes{
 					{
 						Id:          "1",
 						HeadRefName: "spr/master/00000001",
 						BaseRefName: "master",
-						Commits: genclient.PullRequestsRepositoryPullRequestsNodesCommits{
-							Nodes: &genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodes{
+						Commits: genclient.PullRequestsViewerPullRequestsNodesCommits{
+							Nodes: &genclient.PullRequestsViewerPullRequestsNodesCommitsNodes{
 								{
-									genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodesCommit{Oid: "1"},
+									genclient.PullRequestsViewerPullRequestsNodesCommitsNodesCommit{Oid: "1"},
 								},
 							},
 						},
@@ -74,16 +74,16 @@ func TestMatchPullRequestStack(t *testing.T) {
 				{CommitID: "00000002"},
 				{CommitID: "00000003"},
 			},
-			prs: genclient.PullRequestsRepositoryPullRequests{
-				Nodes: &genclient.PullRequestsRepositoryPullRequestsNodes{
+			prs: genclient.PullRequestsViewerPullRequests{
+				Nodes: &genclient.PullRequestsViewerPullRequestsNodes{
 					{
 						Id:          "1",
 						HeadRefName: "spr/master/00000001",
 						BaseRefName: "master",
-						Commits: genclient.PullRequestsRepositoryPullRequestsNodesCommits{
-							Nodes: &genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodes{
+						Commits: genclient.PullRequestsViewerPullRequestsNodesCommits{
+							Nodes: &genclient.PullRequestsViewerPullRequestsNodesCommitsNodes{
 								{
-									genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodesCommit{Oid: "1"},
+									genclient.PullRequestsViewerPullRequestsNodesCommitsNodesCommit{Oid: "1"},
 								},
 							},
 						},
@@ -92,10 +92,10 @@ func TestMatchPullRequestStack(t *testing.T) {
 						Id:          "2",
 						HeadRefName: "spr/master/00000002",
 						BaseRefName: "spr/master/00000001",
-						Commits: genclient.PullRequestsRepositoryPullRequestsNodesCommits{
-							Nodes: &genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodes{
+						Commits: genclient.PullRequestsViewerPullRequestsNodesCommits{
+							Nodes: &genclient.PullRequestsViewerPullRequestsNodesCommitsNodes{
 								{
-									genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodesCommit{Oid: "2"},
+									genclient.PullRequestsViewerPullRequestsNodesCommitsNodesCommit{Oid: "2"},
 								},
 							},
 						},
@@ -132,16 +132,16 @@ func TestMatchPullRequestStack(t *testing.T) {
 		{
 			name:    "RemoveOnlyCommit",
 			commits: []git.Commit{},
-			prs: genclient.PullRequestsRepositoryPullRequests{
-				Nodes: &genclient.PullRequestsRepositoryPullRequestsNodes{
+			prs: genclient.PullRequestsViewerPullRequests{
+				Nodes: &genclient.PullRequestsViewerPullRequestsNodes{
 					{
 						Id:          "1",
 						HeadRefName: "spr/master/00000001",
 						BaseRefName: "master",
-						Commits: genclient.PullRequestsRepositoryPullRequestsNodesCommits{
-							Nodes: &genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodes{
+						Commits: genclient.PullRequestsViewerPullRequestsNodesCommits{
+							Nodes: &genclient.PullRequestsViewerPullRequestsNodesCommitsNodes{
 								{
-									genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodesCommit{Oid: "1"},
+									genclient.PullRequestsViewerPullRequestsNodesCommitsNodesCommit{Oid: "1"},
 								},
 							},
 						},
@@ -156,16 +156,16 @@ func TestMatchPullRequestStack(t *testing.T) {
 				{CommitID: "00000001"},
 				{CommitID: "00000002"},
 			},
-			prs: genclient.PullRequestsRepositoryPullRequests{
-				Nodes: &genclient.PullRequestsRepositoryPullRequestsNodes{
+			prs: genclient.PullRequestsViewerPullRequests{
+				Nodes: &genclient.PullRequestsViewerPullRequestsNodes{
 					{
 						Id:          "1",
 						HeadRefName: "spr/master/00000001",
 						BaseRefName: "master",
-						Commits: genclient.PullRequestsRepositoryPullRequestsNodesCommits{
-							Nodes: &genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodes{
+						Commits: genclient.PullRequestsViewerPullRequestsNodesCommits{
+							Nodes: &genclient.PullRequestsViewerPullRequestsNodesCommitsNodes{
 								{
-									genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodesCommit{Oid: "1"},
+									genclient.PullRequestsViewerPullRequestsNodesCommitsNodesCommit{Oid: "1"},
 								},
 							},
 						},
@@ -174,10 +174,10 @@ func TestMatchPullRequestStack(t *testing.T) {
 						Id:          "3",
 						HeadRefName: "spr/master/00000003",
 						BaseRefName: "spr/master/00000002",
-						Commits: genclient.PullRequestsRepositoryPullRequestsNodesCommits{
-							Nodes: &genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodes{
+						Commits: genclient.PullRequestsViewerPullRequestsNodesCommits{
+							Nodes: &genclient.PullRequestsViewerPullRequestsNodesCommitsNodes{
 								{
-									genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodesCommit{Oid: "2"},
+									genclient.PullRequestsViewerPullRequestsNodesCommitsNodesCommit{Oid: "2"},
 								},
 							},
 						},
@@ -186,10 +186,10 @@ func TestMatchPullRequestStack(t *testing.T) {
 						Id:          "2",
 						HeadRefName: "spr/master/00000002",
 						BaseRefName: "spr/master/00000001",
-						Commits: genclient.PullRequestsRepositoryPullRequestsNodesCommits{
-							Nodes: &genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodes{
+						Commits: genclient.PullRequestsViewerPullRequestsNodesCommits{
+							Nodes: &genclient.PullRequestsViewerPullRequestsNodesCommitsNodes{
 								{
-									genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodesCommit{Oid: "2"},
+									genclient.PullRequestsViewerPullRequestsNodesCommitsNodesCommit{Oid: "2"},
 								},
 							},
 						},
@@ -229,16 +229,16 @@ func TestMatchPullRequestStack(t *testing.T) {
 				{CommitID: "00000001"},
 				{CommitID: "00000003"},
 			},
-			prs: genclient.PullRequestsRepositoryPullRequests{
-				Nodes: &genclient.PullRequestsRepositoryPullRequestsNodes{
+			prs: genclient.PullRequestsViewerPullRequests{
+				Nodes: &genclient.PullRequestsViewerPullRequestsNodes{
 					{
 						Id:          "1",
 						HeadRefName: "spr/master/00000001",
 						BaseRefName: "master",
-						Commits: genclient.PullRequestsRepositoryPullRequestsNodesCommits{
-							Nodes: &genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodes{
+						Commits: genclient.PullRequestsViewerPullRequestsNodesCommits{
+							Nodes: &genclient.PullRequestsViewerPullRequestsNodesCommitsNodes{
 								{
-									genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodesCommit{Oid: "1"},
+									genclient.PullRequestsViewerPullRequestsNodesCommitsNodesCommit{Oid: "1"},
 								},
 							},
 						},
@@ -247,10 +247,10 @@ func TestMatchPullRequestStack(t *testing.T) {
 						Id:          "2",
 						HeadRefName: "spr/master/00000002",
 						BaseRefName: "spr/master/00000001",
-						Commits: genclient.PullRequestsRepositoryPullRequestsNodesCommits{
-							Nodes: &genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodes{
+						Commits: genclient.PullRequestsViewerPullRequestsNodesCommits{
+							Nodes: &genclient.PullRequestsViewerPullRequestsNodesCommitsNodes{
 								{
-									genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodesCommit{Oid: "2"},
+									genclient.PullRequestsViewerPullRequestsNodesCommitsNodesCommit{Oid: "2"},
 								},
 							},
 						},
@@ -259,10 +259,10 @@ func TestMatchPullRequestStack(t *testing.T) {
 						Id:          "3",
 						HeadRefName: "spr/master/00000003",
 						BaseRefName: "spr/master/00000002",
-						Commits: genclient.PullRequestsRepositoryPullRequestsNodesCommits{
-							Nodes: &genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodes{
+						Commits: genclient.PullRequestsViewerPullRequestsNodesCommits{
+							Nodes: &genclient.PullRequestsViewerPullRequestsNodesCommitsNodes{
 								{
-									genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodesCommit{Oid: "3"},
+									genclient.PullRequestsViewerPullRequestsNodesCommitsNodesCommit{Oid: "3"},
 								},
 							},
 						},
@@ -314,16 +314,16 @@ func TestMatchPullRequestStack(t *testing.T) {
 				{CommitID: "00000002"},
 				{CommitID: "00000003"},
 			},
-			prs: genclient.PullRequestsRepositoryPullRequests{
-				Nodes: &genclient.PullRequestsRepositoryPullRequestsNodes{
+			prs: genclient.PullRequestsViewerPullRequests{
+				Nodes: &genclient.PullRequestsViewerPullRequestsNodes{
 					{
 						Id:          "1",
 						HeadRefName: "spr/master/00000001",
 						BaseRefName: "master",
-						Commits: genclient.PullRequestsRepositoryPullRequestsNodesCommits{
-							Nodes: &genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodes{
+						Commits: genclient.PullRequestsViewerPullRequestsNodesCommits{
+							Nodes: &genclient.PullRequestsViewerPullRequestsNodesCommitsNodes{
 								{
-									genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodesCommit{Oid: "1"},
+									genclient.PullRequestsViewerPullRequestsNodesCommitsNodesCommit{Oid: "1"},
 								},
 							},
 						},
@@ -332,10 +332,10 @@ func TestMatchPullRequestStack(t *testing.T) {
 						Id:          "2",
 						HeadRefName: "spr/master/00000002",
 						BaseRefName: "spr/master/00000001",
-						Commits: genclient.PullRequestsRepositoryPullRequestsNodesCommits{
-							Nodes: &genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodes{
+						Commits: genclient.PullRequestsViewerPullRequestsNodesCommits{
+							Nodes: &genclient.PullRequestsViewerPullRequestsNodesCommitsNodes{
 								{
-									genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodesCommit{Oid: "2"},
+									genclient.PullRequestsViewerPullRequestsNodesCommitsNodesCommit{Oid: "2"},
 								},
 							},
 						},
@@ -344,10 +344,10 @@ func TestMatchPullRequestStack(t *testing.T) {
 						Id:          "3",
 						HeadRefName: "spr/master/00000003",
 						BaseRefName: "spr/master/00000002",
-						Commits: genclient.PullRequestsRepositoryPullRequestsNodesCommits{
-							Nodes: &genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodes{
+						Commits: genclient.PullRequestsViewerPullRequestsNodesCommits{
+							Nodes: &genclient.PullRequestsViewerPullRequestsNodesCommitsNodes{
 								{
-									genclient.PullRequestsRepositoryPullRequestsNodesCommitsNodesCommit{Oid: "3"},
+									genclient.PullRequestsViewerPullRequestsNodesCommitsNodesCommit{Oid: "3"},
 								},
 							},
 						},

--- a/github/githubclient/gen/genclient/operations.go
+++ b/github/githubclient/gen/genclient/operations.go
@@ -9,19 +9,15 @@ import (
 )
 
 type PullRequestsViewer struct {
-	Login string
+	Login        string
+	PullRequests PullRequestsViewerPullRequests
 }
 
-type PullRequestsRepository struct {
-	Id           string
-	PullRequests PullRequestsRepositoryPullRequests
+type PullRequestsViewerPullRequests struct {
+	Nodes *PullRequestsViewerPullRequestsNodes
 }
 
-type PullRequestsRepositoryPullRequests struct {
-	Nodes *PullRequestsRepositoryPullRequestsNodes
-}
-
-type PullRequestsRepositoryPullRequestsNodes []*struct {
+type PullRequestsViewerPullRequestsNodes []*struct {
 	Id             string
 	Number         int
 	Title          string
@@ -30,31 +26,35 @@ type PullRequestsRepositoryPullRequestsNodes []*struct {
 	HeadRefName    string
 	Mergeable      MergeableState
 	ReviewDecision *PullRequestReviewDecision
-	Repository     PullRequestsRepositoryPullRequestsNodesRepository
-	Commits        PullRequestsRepositoryPullRequestsNodesCommits
+	Repository     PullRequestsViewerPullRequestsNodesRepository
+	Commits        PullRequestsViewerPullRequestsNodesCommits
 }
 
-type PullRequestsRepositoryPullRequestsNodesRepository struct {
+type PullRequestsViewerPullRequestsNodesRepository struct {
 	Id string
 }
 
-type PullRequestsRepositoryPullRequestsNodesCommits struct {
-	Nodes *PullRequestsRepositoryPullRequestsNodesCommitsNodes
+type PullRequestsViewerPullRequestsNodesCommits struct {
+	Nodes *PullRequestsViewerPullRequestsNodesCommitsNodes
 }
 
-type PullRequestsRepositoryPullRequestsNodesCommitsNodes []*struct {
-	Commit PullRequestsRepositoryPullRequestsNodesCommitsNodesCommit
+type PullRequestsViewerPullRequestsNodesCommitsNodes []*struct {
+	Commit PullRequestsViewerPullRequestsNodesCommitsNodesCommit
 }
 
-type PullRequestsRepositoryPullRequestsNodesCommitsNodesCommit struct {
+type PullRequestsViewerPullRequestsNodesCommitsNodesCommit struct {
 	Oid               string
 	MessageHeadline   string
 	MessageBody       string
-	StatusCheckRollup *PullRequestsRepositoryPullRequestsNodesCommitsNodesCommitStatusCheckRollup
+	StatusCheckRollup *PullRequestsViewerPullRequestsNodesCommitsNodesCommitStatusCheckRollup
 }
 
-type PullRequestsRepositoryPullRequestsNodesCommitsNodesCommitStatusCheckRollup struct {
+type PullRequestsViewerPullRequestsNodesCommitsNodesCommitStatusCheckRollup struct {
 	State StatusState
+}
+
+type PullRequestsRepository struct {
+	Id string
 }
 
 // PullRequestsResponse response type for PullRequests
@@ -73,9 +73,6 @@ func (c *gqlclient) PullRequests(ctx context.Context,
 	query PullRequests($repo_owner : String!, $repo_name : String!) {
 		viewer {
 			login
-		}
-		repository(owner: $repo_owner, name: $repo_name) {
-			id
 			pullRequests(first: 100, states: [OPEN]) {
 				nodes {
 					id
@@ -103,6 +100,9 @@ func (c *gqlclient) PullRequests(ctx context.Context,
 					}
 				}
 			}
+		}
+		repository(owner: $repo_owner, name: $repo_name) {
+			id
 		}
 	}`
 

--- a/github/githubclient/queries.graphql
+++ b/github/githubclient/queries.graphql
@@ -4,9 +4,6 @@ query PullRequests(
 ){
 	viewer {
 		login
-	}
-	repository(owner:$repo_owner, name:$repo_name) {
-		id
 		pullRequests(first:100, states:[OPEN]) {
 			nodes {
 				id
@@ -34,6 +31,9 @@ query PullRequests(
 				}
 			}
 		}
+	}
+	repository(owner:$repo_owner, name:$repo_name) {
+		id
 	}
 }
 

